notifications/sinks.go

```diff
@@ -14,6 +14,51 @@ import (
 // externally, but others are suitable for export if the need arises. Albeit,
 // the tight integration with endpoint metrics should be removed.
 
+// SyncBroadcaster will synchronously send events to multiple sinks in serial order
+// If one of the sinks fail the overall operation fails
+type SyncBroadcaster struct {
+	sinks  []Sink
+	closed bool
+}
+
+// NewSyncBroadcaster creates a sink that will write to all the given sinks synchronously in serial order
+func NewSyncBroadcaster(sinks ...Sink) *SyncBroadcaster {
+	return &SyncBroadcaster{
+		sinks: sinks,
+	}
+}
+
+// Write writes the given events synchronously in serial order
+func (b *SyncBroadcaster) Write(events ...Event) error {
+	for _, sink := range b.sinks {
+		if err := sink.Write(events...); err != nil {
+			logrus.WithField("events", events).Errorf(
```

Comment:
Can you use `.WithError`  here?

Reply:
This requires logrus update since current vendored version is old. Let me make another PR to update it.